packages/autorest.python/autorest/preprocess/__init__.py

```diff
@@ -171,6 +171,16 @@ def headers_convert(yaml_data: Dict[str, Any], replace_data: Any) -> None:
             yaml_data[k] = v
 
 
+def has_json_content_type(yaml_data: Dict[str, Any]) -> bool:
+    return any(ct for ct in yaml_data.get("contentTypes", []) if JSON_REGEXP.match(ct))
+
+
+def has_multi_part_content_type(from_typespec: bool, yaml_data: Dict[str, Any]) -> bool:
```

Comment:
nit: a little bit weird to have `from_typespec` here.

Reply:
We don't want to add overload for legacy code which is generated from swagger.

Reply:
I mean not put it in this method, just put the condition in the place call this method.

Reply:
done